src/modules/mavlink/streams/ADSB_VEHICLE.hpp

```diff
@@ -103,6 +128,89 @@ class MavlinkStreamADSBVehicle : public MavlinkStream
 			sent = true;
 		}
 
+#if CONFIG_DRIVERS_TRANSPONDER_UAVIONIX
+
+		if (_mavlink->get_mode() == Mavlink::MAVLINK_MODE_UAVIONIX) {
+
+			// Required update for dynamic message is 5 [Hz]
+			mavlink_uavionix_adsb_out_dynamic_t dynamic_msg = {
+				.utcTime = UINT32_MAX,
+				.gpsLat = INT32_MAX,
+				.gpsLon = INT32_MAX,
+				.gpsAlt = INT32_MAX,
+				.baroAltMSL = INT32_MAX,
+				.accuracyHor = UINT32_MAX,
+				.accuracyVert = UINT16_MAX,
+				.accuracyVel = UINT16_MAX,
+				.velVert = INT16_MAX,
+				.velNS = INT16_MAX,
+				.VelEW = INT16_MAX,
+				.state = UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND,
+				.squawk = static_cast<uint16_t>(_adsb_squawk.get()),
+				.gpsFix = 0,
+				.numSats = UINT8_MAX,
+				.emergencyStatus = static_cast<uint8_t>(_adsb_emergc.get())
+			};
+
+			vehicle_status_s vehicle_status;
+
+			if (_vehicle_status_sub.update(&vehicle_status)) {
+				if (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED) {
+					dynamic_msg.state |= ~UAVIONIX_ADSB_OUT_DYNAMIC_STATE_ON_GROUND;
+				}
+			}
+
+			sensor_gps_s vehicle_gps_position;
+
+			if (_vehicle_gps_position_sub.update(&vehicle_gps_position)) {
+
+				dynamic_msg.utcTime = static_cast<uint32_t>(vehicle_gps_position.time_utc_usec / 1000000ULL);
+				dynamic_msg.gpsLat = vehicle_gps_position.lat;
+				dynamic_msg.gpsLon = vehicle_gps_position.lon;
+				dynamic_msg.gpsAlt = vehicle_gps_position.alt;
+				dynamic_msg.gpsFix = vehicle_gps_position.fix_type;
+				dynamic_msg.numSats = vehicle_gps_position.satellites_used;
+				dynamic_msg.accuracyHor = static_cast<uint32_t>(vehicle_gps_position.eph * 1000.0f); // convert [m] to [mm]
+				dynamic_msg.accuracyVert = static_cast<uint16_t>(vehicle_gps_position.epv * 100.0f); // convert [m] to [cm]
+				//_dynamic_data.accuracyVel //TODO: anything to set here?
+				dynamic_msg.velVert = static_cast<int16_t>(vehicle_gps_position.vel_d_m_s * 100.0f); // convert [m/s] to [cm/s]
```

Comment:
the mavlink message doesn't specify if it's up or down; do we assume it's down because horizontal speed are NE ?

Reply:
I'll check this with uAvionix. Thanks!

Reply:
Answer from uAvionix 
> Climbing would be a positive vertical rate, descending would be negative.

Changed to negative sign after consulting with @bresch